osquery/tables/networking/windows/process_open_sockets.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <ws2tcpip.h>
+#define WIN32_LEAN_AND_MEAN
+#include <Iphlpapi.h>
+#include <windows.h>
+#include <winsock2.h>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+struct Table {
+  size_t numEntries = 0;
+  int protocol = 0;
+  int family = 0;
+  MIB_TCPTABLE_OWNER_PID* tcpTable;
+  MIB_TCP6TABLE_OWNER_PID* tcp6Table;
+  MIB_UDPTABLE_OWNER_PID* udpTable;
+  MIB_UDP6TABLE_OWNER_PID* udp6Table;
+  enum tableType { tcp, tcp6, udp, udp6 } tableType;
+};
+
+void parseSocketTable(QueryData& results, const Table& socketsTable) {
```

Comment:
I usually recommend placing the input variables first, and ending with the mutable/outputs.